servicetalk-concurrent-api/src/test/java/io/servicetalk/concurrent/api/PublisherFlatMapLatestTest.java

```diff
@@ -0,0 +1,355 @@
+                                // the request(n) in onSubscribe without triggering reentry from onNext.
+                                Thread.sleep(100);
+                            } catch (InterruptedException e) {
+                                throw new RuntimeException(e);
+                            }
+                            subscription.request(1);
+                        });
+                    }
+                }
+
+                @Override
+                public void onError(Throwable t) {
+                    signals.add(t);
+                }
+
+                @Override
+                public void onComplete() {
+                    signals.add(completeSignal);
+                }
+            });
+
+            Object signal = signals.take();
+            assertThat(signal, instanceOf(FlatMapLatestSignal.class));
+            validateSignal((FlatMapLatestSignal<Integer>) signal, 100, false);
+            signal = signals.take();
+            assertThat(signal, instanceOf(FlatMapLatestSignal.class));
+            validateSignal((FlatMapLatestSignal<Integer>) signal, 101, false);
+            signal = signals.take();
+            assertThat(signal, instanceOf(FlatMapLatestSignal.class));
+            validateSignal((FlatMapLatestSignal<Integer>) signal, 102, false);
+            signal = signals.take();
+            assertThat(signal, is(completeSignal));
+        } finally {
+            executor.closeAsync().toFuture().get();
+        }
+    }
+
+    @Test
+    void nullPublisher() throws InterruptedException {
+        toSource(publisher.flatMapLatest(toSwitchFunction(i -> i == 0 ? null : Publisher.from("foo"))))
+                .subscribe(subscriber);
+
+        subscriber.awaitSubscription().request(1);
+        testSubscription.awaitRequestN(1);
+
+        publisher.onNext(0);
+
+        testSubscription.awaitRequestN(2);
+        publisher.onNext(1);
+
+        validateSignal(subscriber.takeOnNext(), "foo", false);
+        validateTerminal(publisher, subscriber, false);
+    }
+
+    private static <R> void validateSignal(@Nullable FlatMapLatestSignal<R> signal, R data, boolean isSwitched) {
+        assertThat(signal, notNullValue());
+        assertThat(signal.isSwitched(), equalTo(isSwitched));
+        assertThat(signal.data(), equalTo(data));
+    }
+
+    private static <P, S> void validateTerminal(TestPublisher<P> publisher, TestPublisherSubscriber<S> subscriber,
+                                                boolean onError) {
+        if (onError) {
+            publisher.onError(DELIBERATE_EXCEPTION);
+            assertThat(subscriber.awaitOnError(), is(DELIBERATE_EXCEPTION));
+        } else {
+            publisher.onComplete();
+            subscriber.awaitOnComplete();
+        }
+    }
+
+    /**
+     * A signal containing the data from a series of {@link Publisher}s connected in a serial fashion.
+     * @param <T> Type of the data.
+     */
+    interface FlatMapLatestSignal<T> {
```

Comment:
This has shown to be useful but will keep internal to test for now.

When a switch happens application logic may need to take special action (e.g. replay initial/aggregated state as first event)